src/sieve.py

```diff
@@ -1,20 +1,29 @@
-"""Computing primes."""
-
-
-def sieve(n: int) -> list[int]:
-    """
-    Compute all the primes up to (and possibly including) `n`.
-
-    `n` must be positive.
-
-    >>> sieve(15)
-    [2, 3, 5, 7, 11, 13]
-
-    """
-    assert n > 0
-    candidates = list(range(2, n + 1))
-    primes = []
-
-    # FIXME: fill out this bit
-
-    return primes
+"""Computing primes."""
+
+
+def sieve(n: int) -> list[int]:
+    """
+    Compute all the primes up to (and possibly including) `n`.
+
+    `n` must be positive.
+
+    >>> sieve(15)
+    [2, 3, 5, 7, 11, 13]
+
+    """
+    assert n > 0
+    candidates = list(range(2, n + 1))
+    primes = []
+
+
+    # FIXME: fill out this bit
+
+    for x in candidates:
+        multiple = 2
+        while x * multiple <= max(candidates):
+            if x * multiple in candidates:
+                candidates.remove(x * multiple)
+            multiple += 1

```

Comment:
This solves the problem, but it is a little inefficient. When you write `max(candidates)` it has to run through the entire list, and you do that for each iteration of the inner loop. Likewise, `candidates.remove(x * multiple)` needs to copy a large part of `candidates`, which is also slow, especially in an inner loop.

But, we haven't talked about complexity, so for this week, it is an excellent solution. In one week, try to go back and examine it, and maybe find an even more efficient version.